### ISSUE TITLE
Bugsnag exception handler causes CLI scripts to exit 0 with uncaught exceptions

### ISSUE THREAD

**Issue by blkperl** [bugsnag/bugsnag-php#523]
### Description

Bugsnag exception handler causes CLI scripts to exit 0 with uncaught exceptions

### Issue

Bugsnag exception handler causes CLI scripts to exit 0 with uncaught exceptions

### Environment

* bugsnag-php version: v3.16.0
* PHP version: 7.2
* What web framework are you using, if any?: None
* How are you installing bugsnag-php? (Composer, phar files?): composer
* Operating system name and version: Ubuntu 16.04

#### How to reproduce the issue

```php
<?php
declare(strict_types=1);

require __DIR__ . '/vendor/autoload.php';


$bugsnagClient = \Bugsnag\Client::make('foo');
\Bugsnag\Handler::registerWithPrevious($bugsnagClient);
throw new Exception();
```

```
$ php test.php
$ echo $?
0
```

#### How to workaround the issue

```php
$ cat test.php
<?php
declare(strict_types=1);

require __DIR__ . '/vendor/autoload.php';


$bugsnagClient = \Bugsnag\Client::make('foo');

set_exception_handler(function ($throwable) {
      throw $throwable;
});

\Bugsnag\Handler::registerWithPrevious($bugsnagClient);
throw new Exception();
```

```
$ php test.php
PHP Fatal error:  Uncaught Exception in test.php:14
Stack trace:
#0 {main}
  thrown in test.php on line 14

Fatal error: Uncaught Exception in test.php on line 14

Exception:  in test.php on line 14

Call Stack:
    0.0240    2079608   1. Bugsnag\Handler->exceptionHandler() REDACTED/vendor/bugsnag/bugsnag/src/Handler.php:0
    0.0282    2200056   2. call_user_func:{REDACTED/vendor/bugsnag/bugsnag/src/Handler.php:155}() REDACTED/vendor/bugsnag/bugsnag/src/Handler.php:155
    0.0282    2200056   3. {closure:test.php:9-11}() REDACTED/vendor/bugsnag/bugsnag/src/Handler.php:155

$ echo $?
255
```

**Comment by glennpratt** [comment]
I have the same issue, this has now bitten me multiple times.

**Comment by xander-jones** [comment]
Hi @glennpratt, have you tried using the workaround in @blkperl's issue report using `set_exception_handler()`? You may also be interested in reading the comments in #475 as to why this is happening. --Xander

**Comment by glennpratt** [comment]
Yes @xander-jones thanks. I have read all the issues, though the contention that this is normal behavior is rather off to me. I've been working with PHP for a very long time and maintaining sane behavior when settings these handlers has always seemed a given to me.

Here's yesterdays banging my head against a wall and implementing this in another project:

```php
// Workaround https://github.com/bugsnag/bugsnag-php/issues/523 which causes all PHP
// exceptions to be swallowed and exit 0.
$previousHandler = null;
$previousHandler = set_exception_handler(function ($throwable) use (&$previousHandler): void {
    if ($previousHandler !== null) {
        call_user_func($previousHandler, $throwable);
    }
    throw $throwable;
});
Bugsnag\Handler::registerWithPrevious(self::$bugsnagClient);
```

**Comment by imjoehaines** [comment]
This should now be fixed in [v3.23.1](https://github.com/bugsnag/bugsnag-php/releases/tag/v3.23.1)

**Comment by glennpratt** [comment]
Thanks @imjoehaines!
